src/platform/android/BUILD.gn

```diff
@@ -83,14 +84,20 @@ static_library("android") {
   ]
 
   deps = [
-    ":tracing",
     "${chip_root}/src/app:app_config",
     "${chip_root}/src/app/common:ids",
     "${chip_root}/src/lib/dnssd:platform_header",
     "${chip_root}/src/setup_payload",
   ]
 
-  public_deps = [ "${chip_root}/src/platform:platform_base" ]
+  if (matter_enable_tracing_support) {
+    deps += [ ":tracing" ]
+  }
+
+  public_deps = [
+    "${chip_root}/src/platform:platform_base",
```

Comment:
Is there a reason we need platform-base?
Can these be "deps" instead of "public_deps" (given that changes occur in cpp only, so we do not need to expose these beyond that in headers)?

Reply:
`platform_base` was here before my changes. I've just add `tracing_buildconfig` here.